homeassistant/components/tibber/__init__.py

```diff
@@ -40,9 +43,16 @@
 
     try:
         await tibber_connection.update_info()
-    except asyncio.TimeoutError as err:
-        _LOGGER.error("Timeout connecting to Tibber: %s ", err)
-        return False
+    except asyncio.TimeoutError:
+        _LOGGER.warning("Timeout connecting to Tibber. Will retry in %ss", retry_delay)
+
+        async def retry_setup(now):
+            """Retry setup if a timeout happens on Tibber API."""
+            await async_setup(hass, config, retry_delay=min(2 * retry_delay, 300))
```

Comment:
Maybe put the max interval at 900s?
```suggestion
            await async_setup(hass, config, retry_delay=min(2 * retry_delay, 900))
```